### DISCUSSION TITLE
Update from 8.5.1 to latest (14.1.0) on Gosund SP112

### DISCUSSION THREAD

**Discussion by NetRat88** [arendst/Tasmota#21616] [Support request] [answered]
Dear community,

Sometimes the USB ports on my Gosund SP112 randomly toogle on/off and I would like to update to the latest version from 8.5.1 to 14.1.0.

Am I right, that the migration path from 8.5.1 is first to 9.1.0 and than to 14.1.0 as described [here](https://tasmota.github.io/docs/Upgrading/) and I have first to update with the OTA URL http://ota.tasmota.com/tasmota/release-9.1.0/tasmota.bin.gz and than with OTA URL http://ota.tasmota.com/tasmota/release/tasmota.bin.gz?

Or can I simply use the URL http://ota.tasmota.com/tasmota/release/tasmota.bin.gz for the OTA update from version 8.5.1 and it will do the rest automatically?

Many thanks in advance and best regards!

**Comment by arendst** [comment]
As noted in the release notes: First 9.1.0, then latest release.

The random toggle may well be a retained MQTT message....

**Comment by NetRat88** [comment]
Dear community,

today I made the upgrade of my Gosund SP112 from 8.5.1 to 9.1.0 to the latest 14.4.1.
The upgrade was successful but the power statistics in the web UI are missing:

Before Upgrade from 8.5.1. to 14.4.1:
<img width="371" alt="image" src="https://github.com/user-attachments/assets/c06fd495-105b-4246-90be-bec2e679b0af" />

After Upgrade to 14.4.1:
<img width="395" alt="image" src="https://github.com/user-attachments/assets/1012c2ca-6d12-4a16-8656-1f6ad58be8f8" />

Found this discussion (https://github.com/arendst/Tasmota/discussions/13737) and thats why I tried different Templates without any success and hope you can help to solve the problem.
The template I used before the upgrade:

{"NAME":"SP112new","GPIO":[57,0,56,0,132,134,0,0,131,30,21,0,0],"FLAG":4,"BASE":18}

The templates I tried after the update:

Source 1 (https://templates.blakadder.com/gosund_SP112_v3_4.html):
{"NAME":"Gosund 112v3.4","GPIO":[56,0,57,0,132,134,0,0,131,30,21,0,0],"FLAG":4,"BASE":18}

and

{"NAME":"SP112new","GPIO":[57,0,56,0,132,134,0,0,131,30,21,0,0],"FLAG":4,"BASE":18}

Source 2 (https://templates.blakadder.com/gosund_SP112.html):
{"NAME":"SHP5","GPIO":[57,145,56,146,255,22,0,0,255,255,21,255,17],"FLAG":0,"BASE":18}

Many thanks in advance
NetRat

**Reply by sfromis** [reply, marked as answer]
You need to upgrade from the lite build to the full Tasmota build.

**Reply by NetRat88** [reply]
Oh man what a stupid mistake!!! Thanks a lot and this solved the issue....

**Reply by D3ltorohd** [reply]
> Oh man what a stupid mistake!!! Thanks a lot and this solved the issue....

Did this work for you ? I will update my Gosund SP112 to, im running 8.1.0.2. So what did @sfromis mean with update from lite to full ? That stands by my device : Tasmota 8.1.0.2 by Theo Arends . So how did you update your Device ? Wich url for the update at OTA do you use ? Did you have to put the template again and wich one do you use ? Did you recalibrate your Power with the Commands over Console ?

**Reply by sfromis** [reply]
What I meant is that the screenshot has "release-lite" at the bottom, meaning successful upgrade to 14.4.1 - the smaller lite build with less features is recommended for easier upgrades with fewer steps needed when going through the upgrade flow - but then missing the final upgrade to `tasmota.bin.gz` from current release instead of `tasmota-lite.bin.gz`.

The upgrade flow for ESP8266, each step having links to the recommended binaries, is in the docs:
https://tasmota.github.io/docs/Upgrading/#upgrade-flow
In your case, you should upgrade to 8.5.1, 9.1 and then current release. The saved configuration and calibration is not changed when following the upgrade flow.

**Reply by D3ltorohd** [reply]
In the Web Ui should i change the OTA URL to this ?

http://ota.tasmota.com/tasmota/release-8.5.1/tasmota.bin.gz and after that to the 9.1. And then to

http://ota.tasmota.com/tasmota/release/tasmota.bin.gz

or do i need the link without .gz ?

**Reply by sfromis** [reply]
No, you should copy the very specific URL values linked directly, step by step, in the upgrade flow I  gave a link to. Not find any URL elsewhere, until the last step where the URL does not have a specific binary.

Here your 2nd link should be fine, assuming that you are not wanting to use some other binary, like a language-specific one. But only do this after already having completed to upgrade steps to 8.5.1 and 9.1, using the links from the upgrade flow.

**Reply by D3ltorohd** [reply]
Ah now im right here ? I should use the links here ?
<img width="852" height="292" alt="image" src="https://github.com/user-attachments/assets/6e748069-a27d-4793-8643-8ca1667e1f55" />

Lite 8.5.1 and lite 9.1. And then the latest tasmota.bin ?

**Reply by sfromis** [reply]
Right

**Reply by D3ltorohd** [reply]
Hm i have put the link in the ota section and click upgrade, after the device is back no new fw.
<img width="343" height="171" alt="image" src="https://github.com/user-attachments/assets/b84acf31-6c6f-4ad3-a0e2-824ad35c3056" />

I can do what i want, i download the bin file and load it on the device, nothing happend. After restart, the same old version.

**Reply by sfromis** [reply]
Well, generalized words about what you try to do does not illustrate the very exact steps you followed, like specific values, commands and results. A typical issue is doing some small detail in a different way from what was intended.

Maybe you have a network setup blocking external access.

**Reply by D3ltorohd** [reply]
Hm, maybe the wlan to bad, i have tried view times, now im on 9.1
